android/expoview/src/main/java/versioned/host/exp/exponent/ExponentPackage.java

```diff
@@ -223,6 +226,16 @@ private void addViewManagersFromPackages(ReactApplicationContext reactContext,
   }
 
   private ExpoModuleRegistryAdapter createDefaultModuleRegistryAdapterForPackages(List<Package> packages, List<SingletonModule> singletonModules) {
-    return new ExpoModuleRegistryAdapter(new ReactModuleRegistryProvider(packages, singletonModules));
+    return new ExpoModuleRegistryAdapter(new ReactModuleRegistryProvider(packages, singletonModules, mExperienceId));
+  }
+
+  private String getExperienceIdFormManifest(JSONObject manifest) {
+    try {
+      ExperienceId experienceId = ExperienceId.create(manifest.getString(ExponentManifest.MANIFEST_ID_KEY));
+      return experienceId.get();
+    } catch (JSONException e) {
+      e.printStackTrace();
+      return null;
```

Comment:
Is the `null` safe to use? We should make a conscious decision about nullability of `appId`.